radosgw_agent/worker.py

```diff
@@ -292,7 +344,11 @@ def get_bucket_instance_entries(self, marker, instance):
 
     def inc_sync_bucket_instance(self, instance, marker, timestamp, retries):
         max_marker, entries = self.get_bucket_instance_entries(marker, instance)
-        objects = set([entry.object for entry in entries])
+
+        # regardless if entries are versioned, make sure we filter them
+        entries = [i for i in ifilter(filter_versioned_objects, entries)]
+
+        objects = set([entry for entry in entries])
```

Comment:
ah, right, that makes sense. We will need to make the list of retries (and saving of objects to retry) match up, by storing the versioning fields with them/providing default values for other fields for them